### ISSUE TITLE
[BUG] Intellisense doesn't supply labels with type designations at definition

### ISSUE THREAD

**Issue by Serene-Arc** [James-Yu/LaTeX-Workshop#3339]
### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you still see this issue. [Required]

You still see this issue?: Yes

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before filling an issue.

You visited the wiki?: Yes

### If your issue is with compiling a document (not having to do with finding the root file of a project), check first that you can compile manually.

You can compile a TeX document manually?: Yes

## Describe the bug [Required]

When using a label with a specified type, the Intellisense won't register that label or suggest it in situations where it'd used. 

### To Reproduce

```latex
\documentclass{article}
\usepackage[hidelinks]{hyperref}
\usepackage{cleveref}

\crefname{diagram}{diag.}{diag.}
\Crefname{diagram}{Diagram}{Diagrams}

\begin{document}
    \section{Test}
    \label[diagram]{test_label}

    This is a test with a \cref{test_label}.
\end{document}
```

When the label is defined, it's all fine, but when it comes time to be used in the `\cref` function, it doesn't pop up.

### Expected behavior

Expected behaviour would be the label appearing in the Intellisense suggestions so it can be accessed quickly.

## Logs [Required]

Please paste the whole log messages here, not parts of ones. The log should start with `Initializing LaTeX Workshop`. It is very important to identify problems.

### LaTeX Workshop Output [Required]

```
[12:13:21] Initializing LaTeX Workshop.
[12:13:21] Extension root: /home/serene/.vscode-oss/extensions/james-yu.latex-workshop-8.27.2
[12:13:21] $PATH: /usr/local/bin:/usr/bin:/var/lib/snapd/snap/bin:/usr/local/sbin:/opt/cuda/bin:/opt/cuda/nsight_compute:/opt/cuda/nsight_systems/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/serene/.local/bin/
[12:13:21] $SHELL: /bin/zsh
[12:13:21] $LANG: en_AU.UTF-8
[12:13:21] $LC_ALL: undefined
[12:13:21] process.platform: linux
[12:13:21] process.arch: x64
[12:13:21] vscode.env.appName: VSCodium
[12:13:21] vscode.env.remoteName: undefined
[12:13:21] vscode.env.uiKind: 1
[12:13:21] Workspace for configuration: file:///home/serene/mwe
[12:13:21] editor.acceptSuggestionOnEnter: "off"
[12:13:21] latex-workshop.bind.enter.key: true
[12:13:21] latex-workshop.docker.enabled: false
[12:13:21] latex-workshop.docker.image.latex: ""
[12:13:21] latex-workshop.hover.preview.mathjax.extensions: []
[12:13:21] latex-workshop.intellisense.package.enabled: true
[12:13:21] latex-workshop.intellisense.update.aggressive.enabled: false
[12:13:21] latex-workshop.intellisense.update.delay: 1000
[12:13:21] latex-workshop.latex.autoBuild.run: "never"
[12:13:21] latex-workshop.latex.build.forceRecipeUsage: true
[12:13:21] latex-workshop.latex.outDir: "%DIR%"
[12:13:21] latex-workshop.latex.recipes: [
 {
  "name": "latexmk 🔃",
  "tools": [
   "latexmk"
  ]
 },
 {
  "name": "latexmk (latexmkrc)",
  "tools": [
   "latexmk_rconly"
  ]
 },
 {
  "name": "latexmk (lualatex)",
  "tools": [
   "lualatexmk"
  ]
 },
 {
  "name": "latexmk (xelatex)",
  "tools": [
   "xelatexmk"
  ]
 },
 {
  "name": "pdflatex ➞ bibtex ➞ pdflatex × 2",
  "tools": [
   "pdflatex",
   "bibtex",
   "pdflatex",
   "pdflatex"
  ]
 },
 {
  "name": "Compile Rnw files",
  "tools": [
   "rnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "Compile Jnw files",
  "tools": [
   "jnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "tectonic",
  "tools": [
   "tectonic"
  ]
 }
]
[12:13:21] latex-workshop.latex.tools: [
 {
  "name": "latexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-pdf",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "lualatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-lualatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "xelatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-xelatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "latexmk_rconly",
  "command": "latexmk",
  "args": [
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "pdflatex",
  "command": "pdflatex",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "bibtex",
  "command": "bibtex",
  "args": [
   "%DOCFILE%"
  ],
  "env": {}
 },
 {
  "name": "rnw2tex",
  "command": "Rscript",
  "args": [
   "-e",
   "knitr::opts_knit$set(concordance = TRUE); knitr::knit('%DOCFILE_EXT%')"
  ],
  "env": {}
 },
 {
  "name": "jnw2tex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"tex\")"
  ],
  "env": {}
 },
 {
  "name": "jnw2texmintex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"texminted\")"
  ],
  "env": {}
 },
 {
  "name": "tectonic",
  "command": "tectonic",
  "args": [
   "--synctex",
   "--keep-logs",
   "%DOC%.tex"
  ],
  "env": {}
 }
]
[12:13:21] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto"
[12:13:21] Creating a new file watcher.
[12:13:21] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[12:13:21] Creating PDF file watcher.
[12:13:21] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{}}
[12:13:21] Creating Bib file watcher.
[12:13:21] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[12:13:21] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[12:13:21] [Server] Creating LaTeX Workshop http and websocket server.
[12:13:21] LaTeX Workshop initialized.
[12:13:21] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[12:13:21] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[12:13:21] Current workspace folders: ["file:///home/serene/mwe"]
[12:13:21] Current workspaceRootDir: file:///home/serene/mwe
[12:13:21] [Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":44241}
[12:13:21] Snippet data loaded.
[12:13:21] [Server] valdOrigin is http://127.0.0.1:44241
[12:13:21] Cannot find fls file: /home/serene/mwe/main.fls
[12:13:21] Cannot find fls file: /home/serene/mwe/parts/appendices/main.fls
[12:41:52] onDidSaveTextDocument triggered: file:///home/serene/mwe/main.tex
[12:42:00] onDidSaveTextDocument triggered: file:///home/serene/mwe/main.tex
[12:42:56] onDidSaveTextDocument triggered: file:///home/serene/mwe/main.tex
[12:43:10] onDidSaveTextDocument triggered: file:///home/serene/mwe/main.tex
[12:52:21] LOG command invoked: default
[12:52:35] Current workspace folders: ["file:///home/serene/mwe"]
[12:52:35] Found root file from active editor: /home/serene/mwe/main.tex
[12:52:35] Root file changed: from undefined to /home/serene/mwe/main.tex
[12:52:35] Start to find all dependencies.
[12:52:35] Root file languageId: latex
[12:52:35] Reset file watcher.
[12:52:35] Parsing a file and its subfiles: /home/serene/mwe/main.tex
[12:52:35] Parse fls file.
[12:52:35] Fls file found: /home/serene/mwe/main.fls
[12:52:35] Parse aux file: /home/serene/mwe/main.aux
[12:52:35] Added to file watcher: /home/serene/mwe/main.tex
[12:52:36] Checking for duplicate labels: /home/serene/mwe/main.tex.
[12:52:37] Manager.fileWatcher.getWatched: {"/home/serene/mwe":["main.tex"]}
[12:52:37] Manager.filesWatched: ["/home/serene/mwe/main.tex"]
[12:52:37] BibWatcher.bibWatcher.getWatched: {}
[12:52:37] BibWatcher.bibsWatched: []
[12:52:37] PdfWatcher.pdfWatcher.getWatched: {}
[12:52:37] PdfWatcher.pdfsWatched: []
[12:52:37] PdfWatcher.watchedPdfVirtualUris: []
[12:52:37] PdfWatcher.ignoredPdfUris: []
[12:52:53] onDidSaveTextDocument triggered: file:///home/serene/mwe/main.tex
[12:52:53] File watcher - file changed: /home/serene/mwe/main.tex
[12:52:53] Parsing a file and its subfiles: /home/serene/mwe/main.tex
[12:52:53] Checking for duplicate labels: /home/serene/mwe/main.tex.

```

### Developer Tools Console [Required]

Nothing Suspicious to report

## Desktop [Required]

Please write exact version numbers. Please don't write `latest` instead of exact numbers.

 - OS: Linux (Arch)
 - VS Code version: VS Codium 1.68.1
 - Extension version: 8.27.2
 - TeX distribution version: pdfTeX 3.141592653-2.6-1.40.24 (TeX Live 2022/Arch Linux)

### Are you using VSCodium?

Yes

### Are you using the Snap or Flatpack versions of VS Code?

No

### Are you using LaTeX Workshop with VS Code Remote?

No

### Additional context

Tried to have a crack at it myself but VS Code and Typscript aren't my specialty, sorry.
